lib/Sema/TypeCheckAttr.cpp

```diff
@@ -3081,6 +3082,37 @@ synthesizeMainBody(AbstractFunctionDecl *fn, void *arg) {
   return std::make_pair(body, /*typechecked=*/false);
 }
 
+llvm::SmallString<128> generateMainFunctionText(ASTContext &C, Decl *decl,
+                                                bool isThrows, bool isAsync) {
+  // Prepare the indent (same as `printRequirementStub`)
+  StringRef ExtraIndent;
+  StringRef CurrentIndent = Lexer::getIndentationForLine(
+      C.SourceMgr, decl->getStartLoc(), &ExtraIndent);
```

Comment:
We should pass the sum indent to the printer instead.
```
std::string MethodIndent = (CurrentIndent + ExtraIndent).str();
```

Then you won’t have to explicitly print any indents except for the placeholder’s extra indent.